pkg/portal/model/app.go

```diff
@@ -1,10 +1,27 @@
 package model
 
 import (
+	"io/ioutil"
+
 	"github.com/authgear/authgear-server/pkg/lib/config"
 )
 
 type App struct {
 	ID      string
 	Context *config.AppContext
 }
+
+func (a *App) LoadFile(path string) ([]byte, error) {
```

Comment:
I suggest we change AppContext to include effective config and raw config.

```
type AppContext struct {
  Fs fs.Fs
  RawConfig *Config
  EffectiveConfig *Config
}
```

Reply:
I'd prefer not to do this, as it is not supposed to be used directly, and having same `*Config` type as effective config may allow potential misuse.

I intended to add a `configFile(path: String!)` GraphQL field later on to get template files. In this case, the `rawAppConfig` and `rawSecretConfig` fields would be like a shorthand to that field.

Reply:
You mean?

```
configFile(path: "authgear.yaml")
configFile(path: "authgear.secrets.yaml")
configFile(path: "templates/auth_ui_translation.json")
```

I just realize [alias](https://graphql.org/learn/queries/#aliases)!

Reply:
Anyway we need an extra argument `type: "raw" | "effective"` to fetch raw / effective config.

Reply:
I suppose we can keep the effective app config field, the app node would look like: 
```graphql
type App implements Node {
  id: ID!

  configFile(path: String!): String!
  """ same as configFile(path: "authgear.yaml") """
  rawAppConfig: AppConfig!
  """ same as configFile(path: "authgear.secrets.yaml") """
  rawSecretConfig: SecretConfig!
  effectiveAppConfig: AppConfig!
}
```

Reply:
So effectiveAppConfig cannot be fetched via `configFile()`?

I prefer we remove the shorthand.

Reply:
> So effectiveAppConfig cannot be fetched via configFile()?

Yes, `configFile` field only return the content of the specified file in the FS, without any processing.

> I prefer we remove the shorthand.

Sure, we can remove it. So the frontend need to understand which files to fetch (`authgear.yaml` and `authgear.secrets.yaml`, or other template files), and decode the YAML at browser. Does that sounds good to you?

Reply:
Oh, I didn't realize they are not strictly shorthand because the type of the field is different. Let's keep them